lua/fspectate/sv_init.lua

```diff
@@ -23,12 +23,18 @@ local function findPlayer(info)
     return nil
 end
 
+local FSpectating = {}
+-- For Lua Refresh
+for _,v in ipairs(player.GetHumans()) do
+    FSpectating[v] = v.FSpectating
+end
 local function startSpectating(ply, target)
     local canSpectate = hook.Call("FSpectate_canSpectate", nil, ply, target)
     if canSpectate == false then return end
 
     ply.FSpectatingEnt = target
     ply.FSpectating = true
+    FSpectating[ply] = true
```

Comment:
This tabe will fill up when players disconnect while spectating. Probably best to clear it out once in a while.